src/types.ts

```diff
@@ -5,22 +5,21 @@ import { Flow } from './core/flow';
 import { INVALID_MOVE } from './core/reducer';
 import * as StorageAPI from './server/db/base';
 import { EventsAPI } from './plugins/plugin-events';
-import { PlayerAPI } from './plugins/plugin-player';
 import { RandomAPI } from './plugins/plugin-random';
 
 export { StorageAPI };
 
 export type AnyFn = (...args: any[]) => any;
 
-export interface State {
-  G: object;
-  ctx: Ctx;
+export interface State<G extends any = any, CtxWithPlugins extends Ctx = Ctx> {
+  G: G;
+  ctx: CtxWithPlugins;
```

Comment:
Should we mark this as `Ctx | CtxWithPlugins` in order to emphasize that the state is enhanced with plugins in some contexts, but bare in other?

Reply:
Ooh I like that. I was thinking about that and wasn't sure how we could make it more semantic